src/java_bytecode/java_string_library_preprocess.cpp

```diff
@@ -455,39 +434,23 @@ exprt java_string_library_preprocesst::get_data(
 /// \param symbol_table: symbol table
 /// \param code: code block, in which some assignments will be added
 /// \return a string expression
-string_exprt java_string_library_preprocesst::replace_char_array(
+refined_string_exprt java_string_library_preprocesst::replace_char_array(
   const exprt &array_pointer,
   const source_locationt &loc,
   symbol_tablet &symbol_table,
   code_blockt &code)
 {
-  refined_string_typet ref_type=refined_string_type;
   dereference_exprt array=
     checked_dereference(array_pointer, array_pointer.type().subtype());
-  exprt array_data=get_data(array, symbol_table);
-  // `deref_array` is *(array_pointer->data)`
-  // No null-pointer-exception check here since all array structures
-  // have non-null data
-  const typet &content_type=ref_type.get_content_type();
-  dereference_exprt deref_array(array_data, array_data.type().subtype());
-
+  const exprt array_data=get_data(array, symbol_table);
```

Comment:
const ref?

Reply:
Not possible, symbol table is modified in `get_fresh_aux_symbol`.